datalad/distribution/create.py

```diff
@@ -10,78 +10,178 @@
+            annex""",
             action='store_false'),
         annex_version=Parameter(
             args=("--annex-version",),
-            doc="""select particular annex repository version.  The list of
-            supported versions depends on the available git-annex version""",
+            doc="""select a particular annex repository version. The
+            list of supported versions depends on the available git-annex
+            version. This should be left untouched, unless you know what
+            you are doing""",
             constraints=EnsureDType(int) | EnsureNone()),
         annex_backend=Parameter(
             args=("--annex-backend",),
+            constraints=EnsureStr() | EnsureNone(),
             # not listing choices here on purpose to avoid future bugs
             doc="""set default hashing backend used by the new dataset.
             For a list of supported backends see the git-annex
-            documentation""",
+            documentation. The default is optimized for maximum compatibility
+            of datasets across platforms (especially those with limited
+            path lengths)""",
             nargs=1),
+        git_opts=git_opts,
+        annex_opts=annex_opts,
+        annex_init_opts=annex_init_opts,
     )
 
     @staticmethod
-    @datasetmethod(name='create', dataset_argname='loc')
+    @datasetmethod(name='create', dataset_argname='path')
     def __call__(
-            loc=None, description=None, no_annex=False, annex_version=None,
-            annex_backend='MD5E'):
-        if description and no_annex:
-            raise ValueError("Incompatible arguments: cannot specify description for "
-                             "annex repo and declaring no annex repo.")
-        if loc is None:
-            loc = os.curdir
-        elif isinstance(loc, Dataset):
-            loc = loc.path
-        if no_annex:
-            lgr.info("Creating a new git repo at %s", loc)
-            vcs = GitRepo(loc, url=None, create=True)
+            path=None,
+            description=None,
+            add_to_super=False,
+            no_annex=False,
+            annex_version=None,
+            annex_backend='MD5E',
+            git_opts=None,
+            annex_opts=None,
+            annex_init_opts=None):
+            # TODO: name? (technically not equal to path)
+
+        if path:
+            if isinstance(path, Dataset):
+                ds = path
+            else:
+                ds = Dataset(path)  # TODO: Is there a need to resolve path?
         else:
-            # always come with annex when created from scratch
-            lgr.info("Creating a new annex repo at %s", loc)
-            vcs = AnnexRepo(
-                loc, url=None, create=True, backend=annex_backend,
-                version=annex_version, description=description)
-        return Dataset(loc)
+            ds = Dataset(getpwd())
+
+        if add_to_super:
+            sds_path = GitRepo.get_toppath(opj(ds.path, os.pardir))
```

Comment:
Wrote a `Dataset().get_superdataset()` method for now. `get_containing_dataset` currently does something slightly different. But we might join both eventually.
As for `opj(..., parent)` : Yes, not nice, but we currently rely on `git rev-parse --show-toplevel` and I think we rightfully do so. Therefore stepping up one level doesn't seem to be avoidable.